### PR TITLE
[GL-3536] solved the errors with lint globbing and updated 

### PR DESCRIPTION
**Pull request by Warrior0078**
1. while the lint was executing it gave an error with prevent spell check with globbing.
2. to solve this i have updated the line which have $symbol with ""

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,13 +7,19 @@ jobs:
...
       DOCKER_IMAGE_GCR: gcr.io/gloot-automation/action-deploy-helm
     steps:
     - name: Checkout
       uses: actions/checkout@v2
       with:
         fetch-depth: 0
+    - name: Lint Code Base
+      uses: github/super-linter/slim@v4
+      env:
+        VALIDATE_ALL_CODEBASE: false
```
Comment (philipgloot) [inline .github/workflows/main.yml]: Can you run it once withouth `VALIDATE_ALL_CODEBASE: false` to test all files

**Review by philipgloot — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -3,7 +3,8 @@ FROM google/cloud-sdk:alpine
 ENV XDG_DATA_HOME=/helm3home
 
 # Install openssl
-RUN apk add --no-cache openssl
+RUN apk add --no-cache openssl=1.1.1b-r1 && \
```
Comment (philipgloot) [inline Dockerfile]: `1.1.1b-r1` is not the latest version

File: value-files/override-new-name.yaml [hunk 3]
```diff
@@ -1,2 +1,2 @@
 application:
-    name: "github-action-helm-test-multiple-files"
\ No newline at end of file
+name: "github-action-helm-test-multiple-files"
```
Comment (philipgloot) [inline value-files/override-new-name.yaml]: fix indentation

File: value-files/old-name.yaml [hunk 4]
```diff
@@ -11,6 +10,6 @@ ports:
     protocol: TCP
 
 probe:
-  httpGet:
-    path: /
-    port: 80
\ No newline at end of file
+    httpGet:
+      path: /
+      port: 80
```
Comment (philipgloot) [inline value-files/old-name.yaml]: Fix indentation

File: deploy.sh [hunk 5]
```diff
@@ -14,24 +15,24 @@ echo "Helm Command: $INPUT_HELM_COMMAND"
...
 log_url="https://console.cloud.google.com/logs/query;query=resource.labels.project_id%3D%22$INPUT_PROJECT%22%0Aresource.labels.location%3D%22$INPUT_ZONE%22%0Aresource.labels.cluster_name%3D%22$INPUT_CLUSTER%22%0A%22$RELEASE_NAME%22;cursorTimestamp=$timestamp?project=$INPUT_PROJECT"
 echo "GCP logs:  $log_url"
 echo "::set-output name=log_url::$log_url"
 
 
-helm_command_array=($(eval "echo $INPUT_HELM_COMMAND"))
-declare -a 'helm_args_array=('"$INPUT_HELM_ARGS"')'
+HELM_COMMAND_ARRAY="$(eval "echo $INPUT_HELM_COMMAND")"
+declare -a 'HELM_ARGS_ARRAY=('("$INPUT_HELM_ARGS")')
 helm "${helm_command_array[@]}" "${helm_args_array[@]}"
```
Comment (philipgloot) [inline deploy.sh]: If changing a variable name it needs to be changed everywhere

File: deploy.sh [hunk 6]
```diff
@@ -14,24 +15,24 @@ echo "Helm Command: $INPUT_HELM_COMMAND"
...
 
 export XDG_DATA_HOME=/helm3home
 helm pull "$INPUT_OCI_HELM_REPO" --version "$INPUT_OCI_CHART_VERSION"
 #helm repo update
-gcloud auth configure-docker 
+gcloud auth configure-docker
 gcloud container clusters get-credentials "$INPUT_CLUSTER" --zone "$INPUT_ZONE" --project "$INPUT_PROJECT"
 
-export RELEASE_NAME=$(/applicationName.sh "$INPUT_HELM_ARGS")
+readonly export RELEASE_NAME=$(/applicationName.sh "$INPUT_HELM_ARGS")
```
Comment (philipgloot) [inline deploy.sh]: Look here how to set readonly variables https://unix.stackexchange.com/questions/66627/is-there-anyway-to-set-a-readonly-environment-variable#:~:text=readonly%20VAR%3Dfoo%20%20%20%23%20POSIX

`readonly VAR=foo`

File: deploy.sh [hunk 7]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
+# shellcheck disable=SC1072,SC1073,SC1064,SC1078,SC1009,SC1079
 set -e
-export REPOSITORY=$(eval "REPO=$INPUT_REPOSITORY" && echo $REPO | sed 's/\//__/') # Replace '/' with '__', since labels don't support '/'
+export REPOSITORY="$(eval "REPO=$INPUT_REPOSITORY" && echo $REPO" | sed 's/\//__/') # Replace '/' with '__', since labels don't support '/'
```
Comment (philipgloot) [inline deploy.sh]: the second " doesnt seem to be placed correct

File: applicationName.sh [hunk 8]
```diff
@@ -1,59 +1,46 @@
...
+if [ ! "$OLD_NAME" = "null" ]; then
+  export APPLICATION_NAME="$OLD_NAME"
+elif [ ! "$NEW_NAME" = "null" ]; then
+  export APPLICATION_NAME="$NEW_NAME"
 else
   echo "NO APPLICATION_NAME :("
-  exit 1
-fi
-
-echo $APPLICATION_NAME
```
Comment (philipgloot) [inline applicationName.sh]: Why is this deleted?

File: applicationName.sh [hunk 9]
```diff
@@ -1,59 +1,46 @@
...
-then
-  REVERSED_FILES=$(echo $FILES | awk '{ for (i=NF; i>1; i--) printf("%s ",$i); print $1; }')
-  yq m $REVERSED_FILES > "tmp.yaml"
+if [ ${#FILEARRAY[@]} -gt 1 ]; then
+  REVERSED_FILES="$(echo "$FILES" | awk '{ for (i=NF; i>1; i--) printf("%s ",$i); print $1; }')"
+  yq m "$REVERSED_FILES" >"tmp.yaml"
   VALUES="tmp.yaml"
 else
   VALUES=$FILES
-fi
```
Comment (philipgloot) [inline applicationName.sh:43]: is this suppose to be deleted?

File: applicationName.sh [hunk 10]
```diff
@@ -1,59 +1,46 @@
...
+  elif [[ "$value" == applicationName=* ]]; then
+    export readonly APPLICATION_NAME=$(echo "$value" | sed 's/applicationName=//g')
+    echo "$APPLICATION_NAME"
     exit 0
-  elif [[ "$value" == application.name=* ]]
-  then
-    export APPLICATION_NAME=$(echo "$value" | sed 's/application.name=//g')
-    echo $APPLICATION_NAME
+  elif [[ "$value" == application.name=* ]]; then
+    export readonly APPLICATION_NAME=$(echo "$value" | sed 's/application.name=//g')
```
Comment (philipgloot) [inline applicationName.sh]: Look here how to set readonly variables https://unix.stackexchange.com/questions/66627/is-there-anyway-to-set-a-readonly-environment-variable#:~:text=readonly%20VAR%3Dfoo%20%20%20%23%20POSIX

`readonly VAR=foo
`

**Review by philipgloot — CHANGES_REQUESTED** [review]

**Comment by Warrior0078** [comment]
CI deploy has been passed with the new openssl version and checked all the lint errors

File: value-files/old-name.yaml [hunk 11]
```diff
@@ -11,6 +10,6 @@ ports:
     protocol: TCP
 
 probe:
-  httpGet:
+    httpGet:
     path: /
-    port: 80
\ No newline at end of file
+    port: 80
```
Comment (philipgloot) [inline value-files/old-name.yaml]: indentation is wrong

File: value-files/override-new-name.yaml [hunk 12]
```diff
@@ -1,2 +1,2 @@
 application:
-    name: "github-action-helm-test-multiple-files"
\ No newline at end of file
+  name: "github-action-helm-test-multiple-files"
```
Comment (philipgloot) [inline value-files/override-new-name.yaml]: indentation is wrong

**Review by philipgloot — CHANGES_REQUESTED** [review]

File: value-files/override-new-name.yaml [hunk 13]
```diff
@@ -1,2 +1,2 @@
 application:
-    name: "github-action-helm-test-multiple-files"
\ No newline at end of file
+    name: "github-action-helm-test-multiple-files"
```
Comment (philipgloot) [inline value-files/override-new-name.yaml]: 2 spaces instead of 4

**Review by philipgloot — CHANGES_REQUESTED** [review]

File: deploy.sh [hunk 14]
```diff
@@ -14,24 +15,22 @@ echo "Helm Command: $INPUT_HELM_COMMAND"
 echo "Helm Args: $INPUT_HELM_ARGS"
 echo "-----"
 
-echo "$INPUT_GCP_KEY" | base64 -d > /tmp/google_credentials.json
+echo "$INPUT_GCP_KEY | base64 -d > /tmp/google_credentials.json"
```
Comment (philipgloot) [inline deploy.sh]: " " is wrong

**Review by philipgloot — CHANGES_REQUESTED** [review]

**Comment by esmaeilram** [comment]
we don't use this PR anymore

**Pull request closed without merging**